### PR TITLE
Add abfss to the list of cloud protocols

### PR DESCRIPTION
**Pull request by philomine**
## Description

This PR adds `abfss` to the list of cloud protocols in kedro/io/core.py. This enables users to open an abfss path.

## Development notes

Tried to load an instance of `kedro.extras.datasets.pandas.ExcelDataSet`: 
- on the current release: doesn't work
- with this fix: works

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes

Unsure about the way this should be documented and/or tested.

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1348"></a>

### CODE REVIEW

**Comment by datajoely** [comment]
Hi @philomine thanks for raising this. @MerelTheisenQB, @idanov I can't shake the feeling we should be more flexible here, we need this change because currently you'll get the following behaviour:

image
 
Is there any merit on making the default case less destructive? It feels like we'll always be playing catch up with fsspec as vendors introduce new protocols

File: RELEASE.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+# Release 0.17.9
+
+## Major features and improvements
+* Added abfss to list of cloud protocols, enabling abfss paths.
```
Comment (lorenabalan) [inline RELEASE.md:4]: We don't have a 0.17.8 release either, could we put these below one section please? :)

**Review by lorenabalan — APPROVED** [review]

**Comment by lorenabalan** [comment]
> Is there any merit on making the default case less destructive? It feels like we'll always be playing catch up with fsspec as vendors introduce new protocols

This code was introduced on purpose because `fsspec` wasn't parsing the Azure protocol correctly at the time (0.16.x). If this has been corrected, and it parses all protocols correctly (including `http`), we can remove this and go back to relying on `infer_storage_options` from `fsspec`.

**Comment by datajoely** [comment]
Linking to #1344

**Comment by MerelTheisenQB** [comment]
> @MerelTheisenQB do we not want to take @lorenabalan 's suggestion and remove the check on our side and delegate to fsspec entirely?

Ideally we handle it on our side, but I'd suggest doing that in a separate task and not blocking this PR from going in so the `abfss` protocol can already be used in the meantime.

**Comment by philomine** [comment]
Hi! I added the test case suggested. I didn't add any documentation as I don't think it is necessary.

**Comment by MerelTheisenQB** [comment]
> Hi! I added the test case suggested. I didn't add any documentation as I don't think it is necessary.

Thank you! It looks like the `lint` checks are failing. You should be able to fix those by running `make lint` and fix any failures it shows. Let me know if you need help with that.

**Review by MerelTheisenQB — APPROVED** [review]
Thanks for this contribution @philomine ⭐

**Pull request merged**